### PR TITLE
Added initial backend part for user registration

### PR DESCRIPTION
**Pull request by orestch**

### CODE REVIEW

File: moninag/moninag/settings.py [hunk 1]
```diff
@@ -119,3 +136,5 @@
     from .local_settings import *
 except ImportError:
     pass
+
+print (DATABASES)
```
Comment (ghost) [inline moninag/moninag/settings.py]: unnecessary print

File: moninag/registration/forms.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+from django.contrib.auth.forms import UserCreationForm
+from registration.models import CustomUser
+from django.template.loader import render_to_string
+from django.core.mail import EmailMultiAlternatives
```
Comment (hutsulyk) [inline moninag/registration/forms.py]: Reoder imports
```
from django.core.mail import EmailMultiAlternatives
from django.contrib.auth.forms import UserCreationForm
from django.template.loader import render_to_string

from registration.models import CustomUser 
```

File: moninag/registration/forms.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        subject = render_to_string('activation_email_subject.txt',
+                                   ctx_dict)
+        # Email subject *must not* contain newlines
+        subject = ''.join(subject.splitlines())
+
+        html_message = render_to_string(
+            'activation_email.html',
+            ctx_dict
+        )
+        #import pdb; pdb.set_trace()
```
Comment (hutsulyk) [inline moninag/registration/forms.py]: Unnecessary comment

**Review by lisovyk — APPROVED** [review]

File: moninag/registration/models.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from hashlib import md5
+from django.db import models
+from django.contrib.auth.models import AbstractBaseUser
+
+
+class CustomUser(AbstractBaseUser):
+    first_name  = models.CharField(max_length=254, blank=True)
+    second_name = models.CharField(max_length=254, blank=True)
+    email       = models.EmailField(blank=True, unique=True)
```
Comment (hutsulyk) [inline moninag/registration/models.py]: Email should be **required field**, otherwise user could sign up  without entering email.
Make it `blank=False`

**Review by hutsulyk — CHANGES_REQUESTED** [review]
Add email settings in **README.md** which are necessary in **local_settings.py**

- Email settings

```
  ACCOUNT_ACTIVATION_DAYS = 7
  EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
  EMAIL_PORT = 587
  EMAIL_HOST_USER = 'moninaginfo@gmail.com'
  EMAIL_HOST_PASSWORD = '1234rewqasdfvcxz'
  EMAIL_USE_TLS = True
  EMAIL_HOST = 'smtp.gmail.com'
  DEFAULT_FROM_EMAIL = 'moninaginfo@gmail.com'
  WSGI_APPLICATION = 'moninag.wsgi.application'
  ```

File: moninag/registration/views.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
+from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from django.contrib import auth
+from django.template.context_processors import csrf
+from moninag.settings import DEFAULT_FROM_EMAIL, DEFAULT_HOST
```
Comment (mhumetskyi) [inline moninag/registration/views.py]: Change `DEFAULT_HOST` to `EMAIL_HOST`

File: moninag/registration/views.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            user_instance = form.save()
+            username = request.POST['email']
+            password = request.POST['password1']
+            user = auth.authenticate(username=username,password=password)
+            #The user is not active until they activate their account through email
+            user.is_active = False
+            user_id = user.id
+            email = user.email
+            user.save()
+            send_activation_email(DEFAULT_HOST, DEFAULT_FROM_EMAIL, email, user_id)
```
Comment (mhumetskyi) [inline moninag/registration/views.py]: Change `DEFAULT_HOST` to `EMAIL_HOST`

File: moninag/registration/admin.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (ghost) [inline moninag/registration/admin.py]: delete default comment

File: moninag/registration/models.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    is_active    = models.BooleanField(default=False)
+    is_admin     = models.BooleanField(default=False)
+    is_staff     = models.BooleanField(default=False)
+    is_superuser = models.BooleanField(default=False)
+    avatar       =  models.CharField(default='', max_length=1000, editable=False)
```
Comment (ghost) [inline moninag/registration/models.py]: this probably doesn't fit PEP8 (multiple spaces before operator)

File: moninag/moninag/settings.py [hunk 9]
```diff
@@ -69,15 +74,17 @@
...
 DATABASES = {
     'default': {
-        'ENGINE': '',
-        'NAME': '',
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'moninag',
+        'USER': 'moninag',
+        'PASSWORD': '12345678',
+        'HOST': 'localhost',
+        'PORT': ''
```
Comment (hutsulyk) [inline moninag/moninag/settings.py]: Empty  default database settings

File: moninag/templates/registration/register.html [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  {% csrf_token %}
+  {{form.as_p}}
+  
+  
+  
+  <input type="submit" value="Register" />
+    
+  </form>
+  
+{% endblock %}
```
Comment (hutsulyk) [inline moninag/templates/registration/register.html]: Add end line

**Review by hutsulyk — APPROVED** [review]

File: moninag/registration/models.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+class CustomUser(AbstractBaseUser):
+    first_name = models.CharField(max_length=254, blank=True)
+    second_name = models.CharField(max_length=254, blank=True)
+    email = models.EmailField(blank=False, unique=True)
+    is_active = models.BooleanField(default=False)
+    is_admin = models.BooleanField(default=False)
+    is_staff = models.BooleanField(default=False)
+    is_superuser = models.BooleanField(default=False)
+    avatar =  models.CharField(default='', max_length=1000, editable=False)
```
Comment (HenrikhB) [inline moninag/registration/models.py]: multiple spaces after operator

File: moninag/registration/views.py [hunk 12]
```diff
@@ -0,0 +1,78 @@
+from django.shortcuts import render
+from django.http import HttpResponseRedirect, JsonResponse
+from django.contrib import auth
+from django.template.context_processors import csrf
+from moninag.settings import DEFAULT_HOST, DEFAULT_FROM_EMAIL
+from registration.utils.send_email import send_activation_email
+from registration.forms import CustomUserCreationForm
+from registration.models import CustomUser
+
+def login(request):
```
Comment (HenrikhB) [inline moninag/registration/views.py]: expected 2 blank lines, found 1

**Review by hutsulyk — COMMENTED** [review]
User sign up, activation, sign in, gravatar integration, work good, and all migrations applied without conflicts! :+1:

File: moninag/registration/models.py (hunk 11 above)
Comment (ghost) [inline moninag/registration/models.py]: multiple spaces after operator!

File: moninag/registration/views.py [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+    c = {}
+    c.update(csrf(request))
+    return render(request, 'login.html', c)
+
```
Comment (ghost) [inline moninag/registration/views.py:15]: expected 2 blank lines, found 1

File: moninag/registration/views.py (hunk 12 above)
Comment (ghost) [inline moninag/registration/views.py:11]: expected 2 blank lines, found 1

File: moninag/registration/views.py [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    activation_key = int(activation_key)
+    user = CustomUser.objects.get(id = activation_key)
+    if not user.is_active:
+        user.is_active=True
+        user.save()
+    else:
+        return render(request, 'already_activated.html')
+    return render(request, 'activate.html')
+
+def auth_view(request):
```
Comment (ghost) [inline moninag/registration/views.py:28]: expected 2 blank lines, found 1

File: moninag/registration/views.py [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+from registration.models import CustomUser
+
+def login(request):
+    c = {}
+    c.update(csrf(request))
+    return render(request, 'login.html', c)
+
+def activate(request, activation_key):
+    activation_key = int(activation_key)
+    user = CustomUser.objects.get(id = activation_key)
```
Comment (ghost) [inline moninag/registration/views.py]: unexpected spaces around keyword / parameter equals

**Review by ghost — CHANGES_REQUESTED** [review]
moninag/registration/views.py doesn't fit PEP8 - reformat code (~10 issues)

**Review by ghost — APPROVED** [review]

**Pull request merged**